### PR TITLE
Move assignments out of asserts.

### PR DESCRIPTION
**Pull request by nirbhayc**

### CODE REVIEW

File: berkdb/rep/rep_lc_cache.c [hunk 1]
```diff
@@ -107,8 +107,8 @@ free_ent(DB_ENV *dbenv, LC_CACHE_ENTRY * e)
 		LC_CACHE_ENTRY *fnd;
 
 		/* XXX remove this if you see it */
-		assert((fnd =
-			hash_find(dbenv->lc_cache.txnid_hash, &e->txnid)) == e);
+		fnd = hash_find(dbenv->lc_cache.txnid_hash, &e->txnid);
```
Comment (adizaimi) [inline berkdb/rep/rep_lc_cache.c]: we can remove fnd totally as it is not used later.

File: berkdb/rep/rep_record.c [hunk 2]
```diff
@@ -3746,9 +3746,8 @@ __rep_process_txn_int(dbenv, rctl, rec, ltrans, maxlsn, commit_gen, lockid, rp,
 	if (txn_rl_args) {
 		int cmp;
 		if (txn_rl_args->lflags & DB_TXN_LOGICAL_BEGIN) {
-			assert((cmp =
-				log_compare(&txn_rl_args->begin_lsn,
-				    &lc.array[0].lsn)) <= 0);
+			cmp = log_compare(&txn_rl_args->begin_lsn, &lc.array[0].lsn);
```
Comment (adizaimi) [inline berkdb/rep/rep_record.c]: cmp is local and not used later, should remove cmp.

File: berkdb/rep/rep_record.c [hunk 3]
```diff
@@ -4408,9 +4407,8 @@ bad_resize:	;
 
 	if (txn_rl_args) {
 		if (txn_rl_args->lflags & DB_TXN_LOGICAL_BEGIN) {
-			assert((cmp =
-				log_compare(&txn_rl_args->begin_lsn,
-				    &rp->lc.array[0].lsn)) <= 0);
+			cmp = log_compare(&txn_rl_args->begin_lsn, &rp->lc.array[0].lsn);
```
Comment (adizaimi) [inline berkdb/rep/rep_record.c]: cmp can be eliminated.

File: berkdb/txn/txn.c [hunk 4]
```diff
@@ -749,7 +749,8 @@ __txn_allocate_ltrans(dbenv, ltranid, begin_lsn, rlt)
 
 	pthread_mutex_lock(&dbenv->ltrans_hash_lk);
 #ifdef LTRANS_DEBUG
-	assert(!(hflt = hash_find(dbenv->ltrans_hash, lt)));
+	hflt = hash_find(dbenv->ltrans_hash, lt);
```
Comment (adizaimi) [inline berkdb/txn/txn.c]: same with hflt, can eliminate.

**Comment by nirbhayc** [comment]
Looking closely, IMO it wouldn't hurt to keep them.

**Pull request closed without merging**